pkg/client/services/redhatopenshift/mgmt/2022-09-04/redhatopenshift/models.go

```diff
@@ -853,12 +1151,908 @@ func (r Resource) MarshalJSON() ([]byte, error) {
 	return json.Marshal(objectMap)
 }
 
-// ServicePrincipalProfile servicePrincipalProfile represents a service principal profile.
-type ServicePrincipalProfile struct {
-	// ClientID - The client ID used for the cluster.
-	ClientID *string `json:"clientId,omitempty"`
-	// ClientSecret - The client secret used for the cluster.
-	ClientSecret *string `json:"clientSecret,omitempty"`
+// Secret secret represents a secret.
+type Secret struct {
+	autorest.Response `json:"-"`
+	// SecretProperties - The Secret Properties
+	*SecretProperties `json:"properties,omitempty"`
+	// ID - READ-ONLY; Fully qualified resource ID for the resource. Ex - /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}
+	ID *string `json:"id,omitempty"`
+	// Name - READ-ONLY; The name of the resource
+	Name *string `json:"name,omitempty"`
+	// Type - READ-ONLY; The type of the resource. E.g. "Microsoft.Compute/virtualMachines" or "Microsoft.Storage/storageAccounts"
+	Type *string `json:"type,omitempty"`
+	// SystemData - READ-ONLY; Azure Resource Manager metadata containing createdBy and modifiedBy information.
+	SystemData *SystemData `json:"systemData,omitempty"`
+}
+
+// MarshalJSON is the custom marshaler for Secret.
+func (s Secret) MarshalJSON() ([]byte, error) {
+	objectMap := make(map[string]interface{})
+	if s.SecretProperties != nil {
+		objectMap["properties"] = s.SecretProperties
+	}
+	return json.Marshal(objectMap)
+}
+
+// UnmarshalJSON is the custom unmarshaler for Secret struct.
+func (s *Secret) UnmarshalJSON(body []byte) error {
+	var m map[string]*json.RawMessage
+	err := json.Unmarshal(body, &m)
+	if err != nil {
+		return err
+	}
+	for k, v := range m {
+		switch k {
```

Comment:
should we consider forcing lowercase here to remove edge cases of different casing coming in from the client?

Reply:
this is all generated code